TCAT/Controllers/HomeViewController.swift

```diff
@@ -134,54 +132,70 @@ class HomeViewController: UIViewController, UITableViewDelegate, UITableViewData
             searchBar.becomeFirstResponder()
         }
     }
+    
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return isBannerShown ? .lightContent : .default
+    }
 
     @objc func reachabilityChanged(note: Notification) {
         let reachability = note.object as! Reachability
         switch reachability.connection {
             case .none:
-                banner.show(queuePosition: .front, on: self)
-                isBannerShown = true
-                UIApplication.shared.statusBarStyle = .lightContent
+                if !isBannerShown {
+                    banner = StatusBarNotificationBanner(title: "No internet connection", style: .danger)
+                    banner!.autoDismiss = false
+                    banner!.show(queuePosition: .front, on: navigationController)
+                    isBannerShown = true
+                }
                 self.isNetworkDown = true
                 self.sectionIndexes = [:]
                 self.searchBar.isUserInteractionEnabled = false
                 self.sections = []
             case .cellular, .wifi:
                 if isBannerShown {
-                    banner.dismiss()
+                    banner?.dismiss()
+                    banner = nil
```

Comment:
also why are you setting banner to nil and creating a new banner each time (Line 144)? If possible, i think it would be simpler if we created a banner in the beginning and show it and dismiss it depending on the network connection

Reply:
Finnicky banner needs to be nil sometimes to disappear